# Week 01 Question Set

What is your prior experience in this discipline?

No prior experience in sequence analysis. I majored in biology with an emphasis in microbiology/genetics, and feel pretty comfortable with the conceptual part of it. The actual programming, however, is really new to me. I know how to use R for basic statistical analysis but the relationship between github, R, Jupyter, and other programs are going to take a while for me to get used to.

What do you hope to get out of this class?

Mainly, I hope to become comfortable with sequence data. I want to be able to know exactly what to do when I have DNA data. But again, I am new to the world of sequence analysis so I am really just looking forward to learning all that I can.

This class is strongly rooted in an independent project related to genomic analyses. What specific project do you have in mind? If you do not have any data or preference, data can be provided / aquired. If you do not have a specific project, what approach would you like to master as part of this class?

Ingrid mentioned that she talked to her post-doc, Laura, about some projects that I could do for this class. One that I thought might be neat is annotating a snow crab transcriptome from their RNASeq data that might be able to be used for the snow crab gene expression project. If that's something that you think would fit within the scope of this class, then I think that would be a neat project to take over.

What are two things you found most useful from the reading?

I learned (or reaffirmed) the idea that data should be considered read-only. It's important to make new, properly labeled files with any changes to the data instead of altering the original data itself.

I also learned that using a leading 0 for file numbers will maintain the files in the correct order.
